Automattic/jetpack
# Backup: decouple connection screens from useConnection hook

Fixes https://github.com/Automattic/jetpack-backup-team/issues/561

Currently, the `useConnection` hook is used in multiple places on the backup plugin, and it is calling a `backup-promoted-product-info` endpoint anytime the hook is used. Also, we are defining two backup connection screens inside this hook.

The idea is to decouple those components and ensure we are rendering them and calling any API only when it is actually needed.

The proposed approach also fixes https://github.com/Automattic/jetpack-marketing/issues/561. The testimonial carousel is being re-rendered anytime the `useConnection` hook is being used.

## Proposed changes:

* Move `BackupConnectionScreen` and `BackupSecondaryAdminConnectionScreen` into components, including the testimonial images.
  * The images were located on `hooks/assets` are now they are part of `backup-connection-screen/assets`, given they are only used there. 
* Move backup product info API call into a hook
* Refactor `useConnection` to keep it simple: just to return the `connectionStatus` object that comes from `jetpack-connection` package.
* Refactor usage of useConnection hook. Basically, we stop replaced restructured array usage with direct `connectionStatus` assignment.

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-backup-team/issues/561

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:





* Spin up a Jurassic Ninja site using this branch: https://jurassic.ninja/create?jetpack-beta&branches.jetpack-backup=fix/backup-duplicated-calls&wp-debug-log&nojetpack
* Open the developer tools and open the network tab.
* Navigate to the VaultPress Backup page.
* Ensure the `backup-promoted-product-info` is not being called repeatedly.
* Ensure the testimonial section doesn't flicker anymore. [Here is a video of how it is behaving before so you can easily spot the difference](https://github.com/Automattic/jetpack-marketing/issues/561#issuecomment-1953401723).
* Connect your site to Jetpack and purchase a VaultPress Backup plan by clicking on `Get VaultPress Backup`.
* Navigate to the backup admin page
* Ensure there are no attempts on calling the `backup-promoted-product-info` API.
